go-slang/src/typechecker/index.ts

```diff
@@ -282,16 +282,27 @@ export class GolangTypechecker {
       return this.type(astNode.X);
     },
     IfStmt: (astNode: AST.IfStmt) => {
-      return make_undefined_type()
+      return make_undefined_type();
     },
     ForStmt: (astNode: AST.ForStmt) => {
-      return make_undefined_type()
+      return make_undefined_type();
     },
     IncDecStmt: (astNode: AST.IncDecStmt) => {
-      return make_undefined_type()
+      if (astNode.X._type !== AST.NodeType.IDENT)
+        throw new TypeError(
+          `invalid operation ${astNode.Tok} on non-identifier type`,
+        );
+      const ident_type = this.type(astNode.X);
+      if (!is_equal_type(ident_type, make_literal_type(DataType.INT)))
```

Comment:
maybe can refactor out to an `is_int_literal` similar to our `is_number` utils

Reply:
okie, prob refactor in other pr~